ProcessesApi/V1/Factories/ProcessesSnsFactory.cs

```diff
@@ -67,7 +70,41 @@ public EntityEventSns ProcessUpdatedWithMessage(Process process, Token token, st
                 SourceSystem = ProcessUpdatedEventConstants.SOURCE_SYSTEM,
                 EventData = new EventData
                 {
-                    NewData = description
+                    NewData = new Message
+                    {
+                        Description = description
+                    }
+                },
+                User = new User
+                {
+                    Name = token.Name,
+                    Email = token.Email
+                }
+            };
+        }
+
+        public EntityEventSns ProcessUpdatedWithAppointmentRescheduled(Process process, Token token, string oldAppointmentTime, string newAppointmentTime)
```

Comment:
Couldn't you reuse the ProcessUpdated method rather than creating a whole new method? IEntityUpdater should(?) be able to recognise the updates to the appointment datetime.

Reply:
The ProcessUpdated method is used when the [process state is not updated (the process data/Assignment are updated)](https://github.com/LBHackney-IT/processes-api/blob/feature/MTTL-2122/ProcessesApi/V1/Controllers/ProcessesApiController.cs#L168). I did try to reuse this method but it's a bit difficult as it requires for the ProcessState to be of type [UpdateEntityResult](https://github.com/LBHackney-IT/processes-api/blob/feature/MTTL-2122/ProcessesApi/V1/Factories/ProcessesSnsFactory.cs#L119) which then affects the [ConfigureAsync](https://github.com/LBHackney-IT/processes-api/blob/feature/MTTL-2122/ProcessesApi/V1/Services/ProcessService.cs#L60) method. I thought it would be better to have this method rather than having new methods everywhere else. Let me know if you have any suggestions though :)

Reply:
My issue is that we are going to create hundreds of methods if we carry on the way we are, and are creating an event every time the process state is updated. I'm fine with leaving this for now as it may require redoing how we handle processUpdated events in general, just something to highlight.

Reply:
Definitely agree with you. We should update the ProcessUpdatedWithMessage method to be very similar to the ProcessUpdated method, but that would mean we need to change the entire endpoint for all the states we have. Maybe we can do this as a separate task in this sprint or the next.